app/models/ont/library_factory.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+
+# Ont namespace
+module Ont
+  # LibraryFactory
+  # The factory will create libraries in tubes from a given plate
+  class LibraryFactory
+    include ActiveModel::Model
+
+    validate :check_validation_errors,
+             :check_libraries,
+             :check_library_requests,
+             :check_container_materials
+
+    def initialize(attributes = {})
+      @validation_errors = []
+      @libraries = []
+      @library_requests = []
+      @tubes = []
+      @container_materials = []
+
+      return unless fetch_and_validate_entities(attributes)
+
+      num_libraries.times do |lib_idx|
+        build_library(lib_idx, @tag_set.tags.count)
+      end
+    end
+
+    attr_reader :tubes
+
+    def save
+      return false unless valid?
```

Comment:
I think you need to add the `**option` argument to this `save` method and introduce the logic on this line that I used in my other branch.